spec/reek/cli/application_spec.rb

```diff
@@ -0,0 +1,39 @@
+require_relative '../../spec_helper'
+require_lib 'reek/cli/application'
+
+RSpec.describe Reek::CLI::Application do
+  describe '#initialize' do
+    it 'exits with default error code on invalid options' do
+      error_raised = false
+      begin
+        Reek::CLI::Silencer.silently do
+          Reek::CLI::Application.new ['--foo']
+        end
+      rescue SystemExit => e
+        expect(e.status).to eq Reek::CLI::Options::DEFAULT_ERROR_EXIT_CODE
+        error_raised = true
+      end
+      expect(error_raised).to be_truthy
+    end
+  end
+
+  describe '#execute' do
+    let(:command) { double 'reek_command' }
+    let(:app) { Reek::CLI::Application.new [] }
+
+    before do
+      allow(Reek::CLI::ReekCommand).to receive(:new).and_return command
+    end
+
+    it 'executes its command passing itself as a parameter' do
```

Comment:
I'd consider this an implementation detail and rather not have this spec (since refactoring already could break tbis test, which it shouldnt).